projects/step-core/src/lib/client/table/services/table-api-wrapper.service.ts

```diff
@@ -1,19 +1,32 @@
 import { inject, Injectable } from '@angular/core';
 import { TableRequestData } from '../shared/table-request-data';
 import { TableResponseGeneric } from '../shared/table-response-generic';
-import { map, Observable, tap } from 'rxjs';
+import { map, Observable, pipe, tap } from 'rxjs';
 import { AsyncTasksService, AsyncTaskStatusResource, TablesService } from '../../generated';
 import { pollAsyncTask, AsyncTaskStatus } from '../../async-task/async-task.module';
 import { DOCUMENT } from '@angular/common';
+import { HttpClient } from '@angular/common/http';
 
 @Injectable({
   providedIn: 'root',
 })
 export class TableApiWrapperService {
+  private _httpClient = inject(HttpClient);
   private _tables = inject(TablesService);
   private _asyncTaskService = inject(AsyncTasksService);
   private _document = inject(DOCUMENT);
 
+  private proceedAsyncRequest = pipe(
+    pollAsyncTask(this._asyncTaskService),
+    map((status: AsyncTaskStatusResource) => {
+      if (!status?.result?.id) {
+        throw 'Invalid attachment id';
+      }
+      return status.result.id;
+    }),
+    tap((attachmentId) => this.downloadDatasource(attachmentId))
+  );
+
```

Comment:
Doesn't seem to be related to the ticket but I assume it makes sense

Reply:
@jeromecomte During this task implementation, I found out that monitoring screen didn't work because it referenced to legacy export service from angularJs, that was cleaned up from the project.
I fixed it by adding necessary method to this service